### PR TITLE
Add docker-compose stack

### PR DESCRIPTION
**Pull request by nkprince007**
This pull request is a work in progress. Feel free to leave your opinions below.

### CODE REVIEW

**Comment by nkprince007** [comment]
You can locally run from this branch on your machine by using `docker-compose up --build` to build and run the containers. Please check the logs and let me know if I can be of any help.

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    ports:
+      - 80:80
+    volumes:
+      - .:/srv/app
+    environment:
+      DB_HOST: mysql
+      DB_NAME: laravel_docker
+      DB_USER: app
+      DB_PASSWORD: password
+      SYMFONY_ENV: prod
```
Comment (nkprince007) [inline docker-compose.yml]: I still need to provide environment variables for GITLAB_SERVER_URL, GITLAB_TOKEN and SECRET_KEY. I'll do it later after we figure out stuff about the failing app server.

**Comment by nkprince007** [comment]
@lucasaba I did some tinkering and figured out a solution. But I need some help from your side to get the startup script into shape. The containers run `.docker/run.sh` on every run. So, the creation of database and stuff are getting repeated over and over again. Creating database is not required as of now, but schema migrations are something too look forward for.

Is there a way to do proper migrations in symfony? (Like a PHP alternative for `python manage.py migrate`.)

Also, even after running manual migrations, I'm facing a weird 500 error with no logs on apache.

**Comment by lucasaba** [comment]
I've launched the docker image. The first problem I'm facing is that it can't connect to the database.
I've entered on the docker image with this command:
```bash
docker exec -it gitlabtimetrack_app_1 /bin/bash
```
Once inside I can see that:

* environment variables are correctly set
** DB_HOST=mysql
** DB_NAME=gitlab-time-track
** DB_PORT=3306
** DB_PASSWORD=password
** DB_USER=app

But the configuration file app/config/parameters.yml is not correct.
I meen... a user has to edit to fit its need. So, once I've changed it with the correct values:

```yaml
# This file is auto-generated during the composer install
parameters:
    database_host: mysql
    database_port: null
    database_name: gitlab-time-track
    database_user: app
    database_password: password
    secret: ThiSisnOTaSEcreT
    gitlab_token: abcdefghi123455
    gitlab_server_url: 'https://gitlab..com/api/v4/'
```
(almost) everything works.

Regarding the db, since it is build from docker, we can delete the creation command (doctrine:database:create) from .docker/run.sh. And instead of 'creating' the schema, we can just update it:
```bash
php bin/console doctrine:schema:update --no-interaction --force
```
I think that should be all....

**Comment by nkprince007** [comment]
Well, like I mentioned in the readme, you need the whole stack running with compose. Because it is a web app, ideally, we use database inside a container as well. 

```bash
docker-compose up --build
```

**Comment by nkprince007** [comment]
Or, you can run it with your own local database by passing respective environment variables using `-e`.

**Comment by nkprince007** [comment]
With docker, people should not be fiddling around with files. Everything has to be automated.

**Comment by lucasaba** [comment]
There's a problem with file permissions too...I think.

The dockerfile should update some ownership:

```bash
RUN chown -R www-data:www-data /srv/app/cache
RUN chown -R www-data:www-data /srv/app/logs
RUN chown -R www-data:www-data /srv/app/sessions
```

**Comment by nkprince007** [comment]
Yeah, I'm writing a fix right now. Will update it in like 30 min. BRB.

**Comment by nkprince007** [comment]
Changes made. You can test them out by tearing your compose network down and pruning all the volumes. Followed by a fresh re-build.

```bash
docker-compose down -v
docker volume prune
docker-compose up --build
```

**Comment by nkprince007** [comment]
The app is booting up correctly now. But nothing seems to work.

**Comment by nkprince007** [comment]
Sorry, my bad, I used a wrong URL. It works totally fine.

**Comment by lucasaba** [comment]
Wow that's great! I love it ;)
This was a very good lesson form me. Thank you very much ;)

**Pull request merged**